devs-who-run-api/Modules/Auth/Services/AuthService.cs

```diff
@@ -0,0 +1,231 @@
+        var expiresAt = DateTime.UtcNow.AddHours(1);
+
+        var token = new JwtSecurityToken(
+            issuer: configuration["Jwt:Issuer"],
+            audience: configuration["Jwt:Audience"],
+            claims: claims,
+            expires: expiresAt,
+            signingCredentials: credentials
+        );
+
+        var refreshToken = Convert.ToBase64String(RandomNumberGenerator.GetBytes(64));
+
+        return (new JwtSecurityTokenHandler().WriteToken(token), refreshToken, expiresAt);
+    }
+
+    private async Task<string> GetGitHubAccessTokenAsync(string code)
+    {
+        var request = new HttpRequestMessage(HttpMethod.Post, "https://github.com/login/oauth/access_token");
+        var parameters = new Dictionary<string, string>
+        {
+            {"client_id", configuration["GitHub:ClientId"]!},
+            {"client_secret", configuration["GitHub:ClientSecret"]!},
+            {"code", code}
+        };
+
+        request.Content = new FormUrlEncodedContent(parameters);
+        request.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+        var response = await httpClient.SendAsync(request);
+        response.EnsureSuccessStatusCode();
+
+        var result = await response.Content.ReadFromJsonAsync<GitHubTokenResponse>();
+        return result?.AccessToken ?? throw new Exception("Failed to get GitHub access token");
+    }
+
+    private async Task<GitHubUser> GetGitHubUserAsync(string accessToken)
+    {
+        var request = new HttpRequestMessage(HttpMethod.Get, "https://api.github.com/user");
+        request.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", accessToken);
+        request.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+        request.Headers.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("DevsWhoRun", "1.0"));
+
+        var response = await httpClient.SendAsync(request);
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<GitHubUser>() 
+               ?? throw new Exception("Failed to get GitHub user data");
+    }
+
+    private static MemberDTO MapToMemberDto(Member member) => new(
+        member.Id,
+        member.FirstName!,
+        member.LastName!,
+        member.Email!,
+        member.GitHubUserName,
+        member.StravaProfile,
+        member.Avatar,
+        member.UserType
+    );
+}
+
+// Helper classes for GitHub API responses
+public class GitHubTokenResponse
+{
+    public string? AccessToken { get; set; }
+    public string? TokenType { get; set; }
+    public string? Scope { get; set; }
+}
+
+public class GitHubUser
+{
+    public string? Login { get; set; }
+    public string? Id { get; set; }
+    public string? Name { get; set; }
+    public string? Email { get; set; }
+    public string? AvatarUrl { get; set; }
+}
```

Comment:
should we keep this in a separate file?

Reply:
I kept this intentionally for easy reference, will move them into separate file after github auth integration